### ISSUE TITLE
Basic test script isn't working on vscode remote server

### ISSUE THREAD

**Issue by spfrommer** [spfrommer/torchexplorer#53]
Page just shows "Waiting on data..."

Not sure what could be happening here. Will have to debug.

**Comment by farzad-elastix** [comment]
Hi @spfrommer did you figure out what was going on? I see the same issue when I try to visualize Llama through `transformers`.

Here's the code to reproduce:
```python
import transformers
import torchexplorer


model_id = "meta-llama/Llama-3.2-1B-Instruct"

tokenizer = transformers.AutoTokenizer.from_pretrained(model_id)

input_text = "Hello, how are you?"
inputs = tokenizer(input_text, return_tensors="pt")

model: transformers.PreTrainedModel = transformers.AutoModelForCausalLM.from_pretrained(model_id)
model.config.use_cache = False

torchexplorer.watch(model, backend='standalone', log_freq=1)

model(**inputs).logits.sum().backward()
```

**Comment by farzad-elastix** [comment]
Update: I found out that the code only works when model is in `.training` mode. So we can move forward one step by doing `model.train()`, but I run into other errors. I think in my case the issue is that `torchexplorer` doesn't work well when the model has `kwargs` as input. I can start a new issue for this.

**Comment by spfrommer** [comment]
Interestingly, the original script I was trying seems to display totally fine now even on a remote instance. I'm guessing there was a vscode bug which is now fixed.

```python
import torch
import torchvision
import torchexplorer

model = torchvision.models.resnet18(pretrained=False)
dummy_X = torch.randn(5, 3, 32, 32)

torchexplorer.watch(model, log_freq=1, log=['io', 'params'], backend='standalone')
model(dummy_X).sum().backward()
```

But this tool won't be able to handle something as fancy as Llama. It relies on Module inputs / outputs being simple -- it can't handle iterable inputs / outputs (which the HF transformer decoder layers produce). And it certainly can't handle wrapper classes like `CausalLMOutputWithPast`.

But inspecting subcomponents with simple input/output interfaces should work. The following runs for me:
```python
import transformers
import torchexplorer
from transformers import LlamaForCausalLM

model_id = "meta-llama/Llama-3.2-1B-Instruct"

tokenizer = transformers.AutoTokenizer.from_pretrained(model_id)

input_text = "Hello, how are you?"
inputs = tokenizer(input_text, return_tensors="pt")

model: LlamaForCausalLM = transformers.AutoModelForCausalLM.from_pretrained(model_id)
model.config.use_cache = False
model.train()

torchexplorer.watch(model.model.layers[0].mlp, backend='standalone', log_freq=1)

model(**inputs).logits.sum().backward()

```

I'll add a note to the README about this.

**Comment by farzad-elastix** [comment]
The output can easily be turned into a tuple by setting `return_dict=False` but yeah I guess I'm not surprised that the whole model won't work.

Still the output is just a tuple. I think the issue I was seeing was related to input using `kwargs` when trying to visualize the layers (e.g. `model.model.layers[0]`):
```
  File "/Users/farzad/projects/mltest/.venv/lib/python3.13/site-packages/torchexplorer/hook/hook.py", line 129, in pre_hook
    input_gradfns = _get_next_gradfns(inputs)
  File "/Users/farzad/projects/mltest/.venv/lib/python3.13/site-packages/torchexplorer/hook/hook.py", line 342, in _get_next_gradfns
    if tensors[0] is not None and backhook_class in str(tensors[0].grad_fn):
       ~~~~~~~^^^
IndexError: tuple index out of range
```

Thanks for taking a look.

**Comment by spfrommer** [comment]
Gotcha, I'll go ahead and close this issue since it seems like the original problem no longer occurs.
